### ISSUE TITLE
Clear Cell Output removes the PDB interface if active

### ISSUE THREAD

**Issue by stephenholtz** [jupyterlab/jupyterlab#2413]
Using the debugger with e.g. `%pdb on` inside a cell, then clearing the cell output will remove the debugger text entry box. The only way to get it back is to send a keyboard interrupt (stop button) as far as I can tell. Until I realized this, I was restarting the kernel when the debugger output became too long. 

![f4sjwtugbg](https://user-images.githubusercontent.com/1022515/26982663-199bc476-4d07-11e7-9c2c-94fad8721869.gif)

People are probably not using JupyterLab for extensive debugging yet, but it would be nice if on clearing the cell output did not remove the debugger entry box. i.e. If the debugger is active have some command sent to bring back the box.

**Comment by blink1073** [comment]
Oh wow, I just verified that the same happens in the notebook.

**Comment by ellisonbg** [comment]
this may be difficult to fix - the reason is that pdb is using raw input, which in both cases and the message spec is part of output.

**Comment by ellisonbg** [comment]
In the work for #2540 we are removing stdin handling from output in general. We will add it back through another way that doesn't have that problem. Once #2540 is done, we should close this issue and open a new one to track the new implementation of stdin.

**Comment by jasongrout** [comment]
Just curious, can you create a console for the notebook (right-click menu) and interact with pdb via the console?

**Comment by jasongrout** [comment]
Alternatively, perhaps you can start pdb in the console and interact with it from there?

**Comment by mlucool** [comment]
> Just curious, can you create a console for the notebook (right-click menu) and interact with pdb via the console?

No. It freezes there as well if `pdb.set_trace()` is called in the notebook already


> Alternatively, perhaps you can start pdb in the console and interact with it from there?

If `set_trace()` is buried somewhere in another module that makes it harder to use. The interface in console is the same, so if I close the console without cleanly exiting `pdb` I am still in a state that requires me restart the kernel and the notebook is still unusable.

**Comment by jasongrout** [comment]
Thanks for trying. Given that the same problem happens in the classic notebook, apparently, this may need to wait until we refactor the notebook in preparation for real-time collaboration/moving document state serverside (sometime this year?), or for someone to take a look at it and work on a PR. If you'd like to look at it, or know someone that can take a look at it, let us know and we can point you to the relevant files. It sounds like the solution *may* be somewhat tricky.

**Comment by jasongrout** [comment]
Also there is interest from various people for working on debugging capabilities in jlab, which may fill the need here in a more comprehensive way.
